src/charm.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+import logging
+
+from ops.main import main
+from ops.pebble import Layer
+from ops.charm import CharmBase
+from ops.model import ActiveStatus, WaitingStatus, MaintenanceStatus
+
+logger = logging.getLogger(__name__)
+
+
+class TrainingOperatorCharm(CharmBase):
+    """ A Juju Charm for Training Operator"""
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        self.logger = logging.getLogger(__name__)
+
+        self._name = self.model.app.name
+        self._namespace = self.model.name
+        self._manager_service = "manager"
+
+        self.framework.observe(self.on.install, self._on_config_changed)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(self.on.training_operator_pebble_ready, self._on_training_operator_pebble_ready)
+
+    def _training_operator_layer(self):
```

Comment:
This could probably use the `@property` decorator
```suggestion
    @property
    def _training_operator_layer(self) -> dict:
```

Reply:
Is there a preferred method for creating Pebble layers? In the sense that, I could leave it as a dict or use [this](https://github.com/canonical/operator/blob/1.2.0/ops/pebble.py#L445) to instantiate and use the attributes later on, like `service` for this particular charm.

Reply:
Either is fine, using `Layer` is probably preferable, just annotate the return type accordingly